opal/core/search/tests/test_views.py

```diff
@@ -196,6 +197,29 @@ def test_incomplete_matching(self):
         self.assertEqual(data[0]["first_name"], "James")
         self.assertEqual(data[0]["surname"], "Bond")
 
+    def test_number_of_queries(self):
+        """ Pagination should make sure we
+            do the same number of queries
+            despite the number of results.
+        """
+        # we need to make sure we're all logged in before we start
+        self.assertIsNotNone(self.user)
+        for i in range(100):
+            self.create_patient(
+                "James", "Bond", str(i)
+            )
+
+        with self.assertNumQueries(35):
+            self.get_response('{}/?query=Bond'.format(self.url))
```

Comment:
I think that's because of the creation of patient summaries. Which is inefficient because we don't have a db only way to pull our episode start stop, so its manually goes through.

We could probably optimise it, but I don't think those 35 queries are causing us issues now. So they shouldn't cause issues in future. That's the theory anyway.

In fact this *should* (famous last words) be significantly faster, because this will mean that we do the pagination even in the database even when autocomplete search is disabled. So rather than do all patient summary aggregations on every request, we'll only get what we will display.